### PR TITLE
Added Home Tab and Formatted Text Example

### PR DESCRIPTION
**Pull request by rdbreak**
Add home tab example

### CODE REVIEW

File: examples/full.py [hunk 1]
```diff
@@ -3,178 +3,230 @@
...
-from slackify import (ACK, OK, Slackify, async_task, block_reply,
-                      request, respond, text_block, Slack)
-
+from slackify import (
+	ACK, OK, Slackify, async_task, block_reply,
+	request, respond, text_block, reply_text, Slack)
 
 app = Flask(__name__)
-slackify = Slackify(app=app)
+slackify = Slackify(app = app)
```
Comment (gsilvapt) [inline examples/full.py:11]: Python’s [PEP 8 recommendation](https://www.python.org/dev/peps/pep-0008/#other-recommendations) is without spaces in between:

> Don’t use spaces around the = sign when used to indicate a keyword argument, or when used to indicate a default value for an unannotated function parameter:

File: README.md [hunk 2]
```diff
@@ -71,20 +71,11 @@ To connect it to slack you need to meet this preconditions:
...
 
-### Does it support new slack Shorcuts?
-Yes, See [examples/shortcuts.py](examples/shortcuts.py) for a self contained example
-
-### And can i use new slack Modals?
-Of course! See [examples/views.py](examples/views.py) for a quick example
-
-### Are interactive actions supported?
-Yes! See [examples/actions.py](examples/actions.py) for a quickstart.
+### Slackify supports Shortcuts, Slash Commands, Events, Actions, Modals, and Block Kit. See the below examples:
```
Comment (Ambro17) [inline README.md:74]: I would prefer something like 
### Slackify Features
- Slash Commands (See [example](examples/commands.py))
- Shortcuts ...
- ...
So one can have a quick overview of examples with links to the relevant code

File: README.md [hunk 3]
```diff
@@ -71,20 +71,11 @@ To connect it to slack you need to meet this preconditions:
...
+Check the examples directory above for all examples.
 >Note: Legacy actions are unsupported by design as they are discouraged by slack. Nevertheless, if there's popular demand, we could add support for them.
 
-### And slack events?
-As you may have guessed, they are also supported. See [examples/events.py](examples/events.py) for an example.
-
-## Full example
+## Full example (Also in above examples directory)
```
Comment (Ambro17) [inline README.md:78]: I would not clutter the title with this info, perhaps a note at the beginning/end of the example that links to the file on the repo

File: README.md [hunk 4]
```diff
@@ -93,190 +84,10 @@ Here you have a more complete example showcasing all functionality. It includes:
 - A shortcut to roll a dice and get a random number
 - An event handler that echoes reactions to messages.
 - A greeting whenever someone says `hello` in a channel where the bot is present.
+- Home tab example with actions
 >Remember to `export BOT_TOKEN=xoxb-your-bot-secret` to enable slack api calls.
```
Comment (Ambro17) [inline README.md:90]: I would leave the code here too, many mobile users will just scroll the readme without following any link, so it's nice to showcase all features on the main page

File: examples/full.py [hunk 5]
```diff
@@ -3,178 +3,230 @@
...
-                "action_id": NO
-            }
-        ]
-    }
-    blocks = [
-        text_block('Do you like Bots?'),
-        yes_no_buttons_block
-    ]
-    return block_reply(blocks)
+	YES = 'yes'
```
Comment (Ambro17) [inline examples/full.py:17]: Shouldn't this be 4 spaces?

File: examples/full.py [hunk 6]
```diff
@@ -3,178 +3,230 @@
...
+				"value": "i_dont_like_bots",
+				"action_id": NO
+				}
+			]
+		}
+	blocks = [
+		text_block('Do you like Bots?'),
+		yes_no_buttons_block
+		]
+	return block_reply(blocks)
```
Comment (Ambro17) [inline examples/full.py:50]: I think this section might be modified by accident? Could you check if anything changed?
If it's not the case let's leave it as it was regarding spaces

File: examples/full.py [hunk 7]
```diff
@@ -3,178 +3,230 @@
...
 
 
 @slackify.action("yes")
 def yes(action):
-    text_blok = text_block('Super! I do too :thumbsup:')
-    respond(action['response_url'], {'blocks': [text_blok]})
-    return OK
+	text_blok = text_block('Super! I do too :thumbsup:')
```
Comment (Ambro17) [inline examples/full.py:55]: Fix spacing

File: examples/full.py [hunk 8]
```diff
@@ -3,178 +3,230 @@
...
+	respond(action['response_url'], {'blocks': [text_blok]})
+	return OK
 
 
 @slackify.action("no")
 def no(action):
-    text_blok = text_block('Boo! You are so boring :thumbsdown:')
-    respond(action['response_url'], {'blocks': [text_blok]})
-    return OK
+	text_blok = text_block('Boo! You are so boring :thumbsdown:')
```
Comment (Ambro17) [inline examples/full.py:62]: Fix spacing

File: examples/full.py [hunk 9]
```diff
@@ -3,178 +3,230 @@
...
-            "emoji": True
-        },
-        "blocks": modal_blocks
-    }
-    cli.views_open(
-        trigger_id=request.form['trigger_id'],
-        view=registration_form
-    )
-    return OK
+	username_input_block = {
```
Comment (Ambro17) [inline examples/full.py:69]: I think you have configured spacing with tabs, the repo uses 4 spaces instead. Please review changes and revert them so we keep consistency across all files

Perhaps i should add a linter check to avoid this :sweat_smile:

File: examples/full.py [hunk 10]
```diff
@@ -3,178 +3,230 @@
...
+	return OK
 
 
 @slackify.view("registration_form")
 def register_callback(payload):
-    response = payload['view']['state']['values']
-    text_blok = text_block(f':heavy_check_mark: You are now registered.\nForm payload:\n```{response}```')
-    send_message(cli, [text_blok], payload['user']['id'])
-    return ACK
+	response = payload['view']['state']['values']
```
Comment (Ambro17) [inline examples/full.py:137]: fix spaces

File: examples/home_tab.py [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+
+from slackify import Slackify, Flask, Slack, reply_text
+
+# Initialize App
+app = Flask(__name__)
+slackify = Slackify(app = app)
+cli = Slack(os.getenv('SLACK_BOT_TOKEN'))
+
+import os
+from flask import Flask
```
Comment (Ambro17) [inline examples/home_tab.py:13]: Please move all imports at the top level

File: examples/using_blocks.py [hunk 12]
```diff
@@ -0,0 +1,52 @@
+import os
```
Comment (Ambro17) [inline examples/using_blocks.py:1]: We already use blocks on some of the other examples, i.e: `actions.py`.
I'm not sure if it's a great idea to dedicate a whole example just to this, perhaps we could add it to one of the existent files?
How was your experience as a user? Do you feel this will be helpful?

File: examples/views.py [hunk 13]
```diff
@@ -4,101 +4,88 @@
...
+		"type": "modal",
+		"callback_id": "registration_form",
+		"title": {
+			"type": "plain_text",
+			"text": "My First Modal",
+			"emoji": True
+			},
+		"submit": {
+			"type": "plain_text",
+			"text": "Register",
```
Comment (Ambro17) [inline examples/views.py:63]: revert space changes

File: examples/views.py [hunk 14]
```diff
@@ -4,101 +4,88 @@
...
-
-    We'll stick to the first one as it is the easiest for demostration purposes. But
-    read the link to respond by any of the other options.
-    """
-    action = json.loads(request.form["payload"])
-    response = action['view']['state']['values']
-    text_blok = text_block(f':heavy_check_mark: You are now registered.\nPayload:\n```{response}```')
-    send_message(cli, [text_blok], action['user']['id'])
-    return ACK
+	action = json.loads(request.form["payload"])
```
Comment (Ambro17) [inline examples/views.py:82]: Restore docstring and revert unrelated changes

File: examples/views.py [hunk 15]
```diff
@@ -4,101 +4,88 @@
...
+	response = action['view']['state']['values']
+	text_blok = text_block(f':heavy_check_mark: You are now registered.\nPayload:\n```{response}```')
+	send_message(cli, [text_blok], action['user']['id'])
+	return ACK
 
 
 @async_task
 def send_message(cli, blocks, user_id):
-    return cli.chat_postMessage(channel=user_id, user_id=user_id, blocks=blocks)
+	return cli.chat_postMessage(channel = user_id, user_id = user_id, blocks = blocks)
```
Comment (Ambro17) [inline examples/views.py:91]: fix spaces

**Review by Ambro17 — CHANGES_REQUESTED** [review]
Hi @rdbreak! Thank you for your contribution! I had never used the home tab and now we have a nice example on how to do it :smile: 

There are some style changes i would like to adress before merging these. Mainly the use of tabs instead of spaces. I'll probably add a lint check to address this issues earlier so we can focus on code changes on our PRs

Let me know if you have any issues fixing those files

File: examples/full.py (hunk 1 above)
Comment (rdbreak) [reply examples/full.py]: I believe I got this fixed on the last push. Thanks!

File: README.md (hunk 2 above)
Comment (rdbreak) [reply README.md:74]: Good idea! I'll get it fixed

File: README.md (hunk 4 above)
Comment (rdbreak) [reply README.md:90]: Good insight. I'll restore it on the next push

File: examples/full.py (hunk 5 above)
Comment (rdbreak) [reply examples/full.py]: All the spaces issues should be sorted out on the next push 😅

File: examples/full.py (hunk 6 above)
Comment (rdbreak) [reply examples/full.py]: Yes, sorry about that. All the formatting issues should be fixed on the next push

File: examples/full.py (hunk 7 above)
Comment (rdbreak) [reply examples/full.py]: Ack.

File: examples/full.py (hunk 8 above)
Comment (rdbreak) [reply examples/full.py]: Ack.

File: examples/full.py (hunk 9 above)
Comment (rdbreak) [reply examples/full.py]: I have a better idea. I add a linter check so I catch it before pushing it next time? 😂

File: examples/full.py (hunk 10 above)
Comment (rdbreak) [reply examples/full.py]: Ack.

File: examples/home_tab.py (hunk 11 above)
Comment (rdbreak) [reply examples/home_tab.py:2]: Will do.

File: examples/using_blocks.py (hunk 12 above)
Comment (rdbreak) [reply examples/using_blocks.py:1]: If I teleported back to the first time looking at the existing examples, yes. But that's just my opinion, up to you if you want it changed. Let me know

File: examples/views.py (hunk 13 above)
Comment (rdbreak) [reply examples/views.py]: Ack.

File: examples/views.py (hunk 14 above)
Comment (rdbreak) [reply examples/views.py]: Done!

File: examples/views.py (hunk 15 above)
Comment (rdbreak) [reply examples/views.py]: This should be fixed on the next push

File: README.md (hunk 3 above)
Comment (rdbreak) [reply README.md:78]: Will do. I'll revert

**Review by rdbreak — COMMENTED** [review]
I believe all conflicts have been resolved. Please give it a look over when you have a chance

**Comment by rdbreak** [comment]
Looks my IDE settings is causing all these space/tab changes. I'm going to cancel this PR, refork, and resubmit so just the added documents are there. Thanks!
